### PR TITLE
Reworked Teammates page

### PR DESCRIPTION
**Pull request by Twixes**
## Changes

This is a rework of the Teammates page as discussed with @paolodamico and based on https://github.com/PostHog/posthog/issues/2260#issuecomment-723097614. The list of invites stays, but it's consolidated with the list of members now and doesn't have its own page, as its purpose changed to mostly that of reminding about invited users who forgot to join. The overall UX of invites and members should now be clearer.

Teammates

Invite Teammate

## Checklist

- [x] All querysets/queries filter by Organization, Team, and User (if this PR affects ANY querysets/queries).
- [x] Django backend tests (if this PR affects the backend).
- [ ] Cypress end-to-end tests (if this PR affects the frontend).

### CODE REVIEW

**Review by paolodamico — COMMENTED** [review]
It's great that we're making this UX improvement! Won't be able to do a full review of all the functionality or propose more changes right now but here are some suggestions.

- I think it's more common to see team members rather than teammates, at least in American English. Related to this, we also use different terms in some parts of the app (like Organization Members), we should definitely use a single term for everything.
- I'm an admin and I couldn't change the membership level 
    
    - Related to the above, my user is the first user, it should be the owner, but it's not. Not sure if the migration where we introduced these levels considered this.
- Better just to hide the expired invites, doesn't make sense to show them if they're expired. At some point I imagine we'll make a cron job to clean them up anyways.

**Comment by Twixes** [comment]
Thanks.
1. Hm, I think "Team members" is less consistent, since we actually have the Team model… of which "Team Members" are not members – they are members, but of the organization. While "Teammates" is more general. So I'd opt for "Organization Members" when being strict and "Teammates" when being loose.
2. You can't bring other people up to your level for increased security (that is practically only the owner can change other people's access, because the owner can do anything, while admins can't make other people admins…). If this is annoying, we can change it.
3. The expired invite is shown as a reminder to poke that user to finally join (it has a call to action specifically for that reason).
4. BTW, numbered lists of feedback are easier for me to address than plain bullet points. 😂

**Comment by timgl** [comment]
1. Agree it should be team members than teammates, or just "team". 
2. Admins should be able to make other people admin.

File: frontend/src/scenes/organization/Teammates/Members.tsx [hunk 1]
```diff
@@ -12,35 +12,41 @@ import {
...
 ): false | string {
     const currentMembershipLevel = currentOrganization?.membership_level
+    if (memberChanged.user_id === currentUser.id) return "You can't change your own access level."
     if (!currentMembershipLevel) return 'Your membership level is unknown.'
-    if (newLevel) {
-        if (newLevel >= currentMembershipLevel)
+    if (Array.isArray(newLevelOrAllowedLevels)) {
+        if (currentMembershipLevel === OrganizationMembershipLevel.Owner) return false
+        if (!newLevelOrAllowedLevels.length)
+            return "You don't have the permissions to change this member's access level."
```
Comment (timgl) [inline frontend/src/scenes/organization/Teammates/Members.tsx:37]: NIT: "You don't have permission" is more idiomatic english

**Comment by Twixes** [comment]
@timgl 
1. I made the change though I'd still push back against using the word "Team" as it's potentially confusing twofold:
  - We still have the Team (to be: Project) model, of which Team Members are _not_ members – they are Organization Members.
  - We may introduce a _new_ Team model some time, when we need more enterprise capabilities. I'm thinking something like what Sentry has: such a Team would delineate Project access in bigger Organizations, and it would indeed have its Members – but these Team Members would be a different collection than the Team Members (actually Organization Members) proposed here.
2. 👍

**Comment by timgl** [comment]
Conflicts but otherwise looks good

**Pull request merged**